File: examples_ios/MyApp/Sources/SurveyLayouts/DefaultLayout/Items/DefaultLayoutIconLabel.swift
import ConfirmitMobileSDK
import Foundation

class DefaultLayoutIconLabel: DefaultLayoutItem {
  let value: QuestionText
  let iconName: String
  var errors: [ValidationQuestionError]

  init(value: QuestionText, iconName: String) {
    self.value = value
    self.iconName = iconName
    errors = []
  }

  init(errors: [ValidationQuestionError]) {
    value = QuestionText(rawText: "")
    iconName = "outline_sentiment_very_dissatisfied_black_24pt"
    self.errors = errors
  }
}

File: examples_android/app/src/main/java/com/example/myapp/layouts/defaultlayout/items/DefaultLayoutIconLabel.kt
package com.example.myapp.layouts.defaultlayout.items

import com.confirmit.mobilesdk.ui.ValidationQuestionError
import com.confirmit.mobilesdk.ui.questions.QuestionText
import com.example.myapp.R

class DefaultLayoutIconLabel : DefaultLayoutItem {
    val value: QuestionText
    val iconId: Int
    var errors: List<ValidationQuestionError>

    constructor(value: QuestionText, iconId: Int) {
        this.value = value
        this.iconId = iconId
        this.errors = mutableListOf()
    }

    constructor(errors: List<ValidationQuestionError>) {
        this.value = QuestionText("", false)
        this.iconId = R.drawable.ic_sentiment_very_dissatisfied
        this.errors = errors
    }
}